src/rest/server_service/path_predicate.rs

```diff
@@ -0,0 +1,79 @@
+use actix_web::pred::Predicate;
+use actix_web::Request;
+use regex::{self, Regex};
+
+pub struct PathPredicate {
+    regex: Regex,
+}
+
+impl PathPredicate {
+    // The pattern should be a full URI, with or without slash on the beginning.
+    // Wildcards in form of `{some_name}` are allowed, they accept everything except `/`.
+    // The predicate will accept only requests with URIs fully matching the pattern.
+    pub fn for_pattern(pattern: &str) -> Self {
+        let pattern = regex::escape(&pattern);
+        let segment_regex = Regex::new(r#"\\\{[^}]*\}"#).unwrap();
```

Comment:
What should the escaped backslash in the beginning match? The docs do not mention any backslashes in the pattern.

The raw literal could have less line noise. As the pattern does not have any inner double quotes, the `#` guards can be dropped.

Reply:
`{` and `}` are backslash-escaped by `regex::escape

This IS raw literal. I think you've asked me to put all regex patterns into raw literals and I agree that it's a safe habit.

Reply:
Yeah, but it's legal and less noisy to have a `r"foo"` literal if there are no inner double quotes.

Reply:
That's great, I've been always using at least one `#`!